extensions/restart-system-processes/src/index.tsx

```diff
@@ -0,0 +1,266 @@
+
+      resolve(items); // Resolve the promise with the updated items
+    });
+  });
+}
+
+async function getExePath(exe: string) {
+  const path = await new Promise((resolve) => {
+    exec(`which ${exe}`, (error, stdout) => {
+      resolve(error ? null : stdout.trim());
+    });
+  });
+
+  if (path) {
+    return path;
+  }
+
+  // test a few common locations
+  const locations = [`/usr/bin/${exe}`, `/bin/${exe}`, `/usr/sbin/${exe}`, `/sbin/${exe}`];
+  for (const loc of locations) {
+    if (
+      await new Promise((resolve) => {
+        exec(`type ${loc}`, (error) => {
+          resolve(!error);
+        });
+      })
+    ) {
+      return loc;
+    }
+  }
+
+  return null;
+}
+
+export default function Command() {
+  const { push } = useNavigation();
+
+  const [orderedProcesses, setOrderedProcesses] = useState<Process[]>([]);
+  useEffect(() => {
+    orderItems(commonProcesses).then(setOrderedProcesses);
+  }, []);
+
+  return (
+    <List isLoading={orderedProcesses.length === 0}>
+      <List.Section title="Common">
+        {orderedProcesses.map((process) => (
+          <List.Item
+            key={process.process}
+            title={process.label}
+            icon={process.icon}
+            actions={
+              <ActionPanel>
+                <Action
+                  title="Restart"
+                  onAction={async () => {
+                    await performAction({ process });
+                  }}
+                />
+              </ActionPanel>
+            }
+          />
+        ))}
+      </List.Section>
+      {/* This condition needs to be here, otherwise focus would always be on Advanced by default */}
+      {orderedProcesses.length ? (
+        <List.Section title="Advanced">
+          <List.Item
+            key="advanced"
+            title="Advanced Mode..."
+            icon={Icon.BulletPoints}
+            actions={
+              <ActionPanel>
+                <Action
+                  title="View All Running Services"
+                  onAction={async () => {
+                    push(<AdvancedList />);
+                  }}
+                />
+              </ActionPanel>
+            }
+          />
+        </List.Section>
+      ) : null}
+    </List>
+  );
+}
+
+function AdvancedList() {
+  const [items, setItems] = useState<ReactNode[]>([]);
+
+  useEffect(() => {
+    getAdvancedItems().then(setItems);
+  }, []);
+
+  return <List isLoading={items.length === 0}>{items}</List>;
+}
+
+async function performAction(values: { process: Process } | { advancedMode: string }) {
+  const sudoOption = getPreferenceValues<Preferences>().useSudo;
+  const sudo = sudoOption ? "sudo" : "";
+  const processName = "process" in values ? values.process.label : values.advancedMode;
+  let cmd = "";
+
+  if (!process) {
+    await showToast(Toast.Style.Failure, "No process selected");
+    return;
+  }
+
+  await saveLastUsedItems("process" in values ? values.process.process : values.advancedMode);
```

Comment:
this should be moved after the user confirmation, so it only updates the order if the action actually runs